### ISSUE TITLE
device disconnected: the connection has timed out unexpectedly

### ISSUE THREAD

**Issue by rajsite** [daphtdazz/WebBLE#28]
If I run the following page [served from githack](https://raw.githack.com/rajsite/webvi-experiments/d802a60f4913594cdfbec7a42a0afd8dcabc84fb/WebBluetooth/PuckJSBlink.html), click the `Connect to Puck.js` button, and then click `Toggle LED` you can see a connected Puck.js have the LED turn full white and then get the title error message:

```
Device disconnected: The connection has timed out unexpectedly
```

Looking at the [writeValue function in my application](https://github.com/rajsite/webvi-experiments/blob/441daff8cf6d902dfcb8997d0c2655c779d24f51/WebBluetooth/WebBluetooth.gcomp/Support/WebBluetooth.js#L208) I am aware that the value that is passed in is a Uint8Array TypedArray that is a view on a much larger buffer from a WebAssembly module's memory.

If I create a copy of the Uint8Array so that it has it's own much smaller buffer before passing it to WebBLE's implementation of writeValue then no error occurs and the buffer passes through correctly.

My guess is that passing a TypedArray view to the underlying WebAssembly modules memory is not behaving well in the WebBluetooth polyfill implementation.

The following change shows the change to a copy of the buffer that avoids the error: https://github.com/rajsite/webvi-experiments/commit/95fce846e8a49c6940508b1b9c7138ae4857713a

**Comment by daphtdazz** [comment]
Interesting, thanks for reporting it, will have a look when I get a chance.

**Comment by daphtdazz** [comment]
By the way @rajsite, just to confirm, this works fine on e.g. Chrome on the desktop?

**Comment by rajsite** [comment]
Yep yep! It does work without creating a new buffer on latest stable Chrome on Windows Desktop and Android.

**Comment by daphtdazz** [comment]
This was fixed in 1.4.0. However, when using puck js (and probably other Nordic socs) you may find that sending any more than 20 bytes at a time causes a crash, which seems to be a bug in the firmware, which may be fixed in recent updates. But the issue of WebBLE using the underlying ArrayBuffer instead of the TypedArray is fixed (and 1.4.0 has been approved by Apple and is available now on the App Store).

**Comment by rajsite** [comment]
@daphtdazz thanks! I appreciate the update!
